### DISCUSSION TITLE
How long will SQLAlchemy 1.4 be supported?

### DISCUSSION THREAD

**Discussion by jaredmaxwell** [sqlalchemy/sqlalchemy#11452] [Usage Questions] [answered]
Hi,

We are starting to plan high-level tech debt, and we are curious about what window we need to convert our application from 1.4 to 2.0.

I couldn't find any documentation stating when 1.4 or 1.x will stop being supported. Does anyone have a specified EOL?

Thanks!

**Comment by zzzeek** [comment, marked as answer]
maintenance has slowed down on 1.4 to only critical bugs.   when 2.1 comes out, which we are hoping is like a Q1 2025 thing, this will push 1.4 into "near EOL" status, which I only say "near" because we understand a lot of people are still on 1.4 and we will probably still want to backport issue fixes.  however if you're writing any new code at all you'd want to get onto 2.0.   new use cases ettc. are all going to 2.0 only and not 1.4 right now.

**Reply by raylu** [reply]
hello from the last 10 days of Q3 2025. is 1.4 "near EOL" or...?

(also thanks for the detailed migration guide!)

**Reply by CaselIT** [reply]
still near eol, I would expect it to go in "more or less EOL" by the end of the year
